zenoh/src/api/publisher.rs

```diff
@@ -160,28 +163,33 @@ impl<'a> Publisher<'a> {
         }
     }
 
+    #[inline]
     pub fn key_expr(&self) -> &KeyExpr<'a> {
         &self.key_expr
     }
 
+    #[inline]
+    /// Get the `congestion_control` applied when routing the data.
+    pub fn congestion_control(&self) -> CongestionControl {
+        self.congestion_control
+    }
+
     /// Change the `congestion_control` to apply when routing the data.
     #[inline]
     pub fn set_congestion_control(&mut self, congestion_control: CongestionControl) {
         self.congestion_control = congestion_control;
     }
 
-    /// Change the priority of the written data.
+    /// Get the priority of the written data.
     #[inline]
-    pub fn set_priority(&mut self, priority: Priority) {
-        self.priority = priority;
+    pub fn priority(&self) -> Priority {
+        self.priority
     }
 
-    /// Restrict the matching subscribers that will receive the published data
-    /// to the ones that have the given [`Locality`](crate::prelude::Locality).
-    #[zenoh_macros::unstable]
+    /// Change the priority of the written data.
     #[inline]
-    pub fn set_allowed_destination(&mut self, destination: Locality) {
```

Comment:
what happened with `set_allowed_destination`. It it's removal planned?

Reply:
I discussed about it with @OlivierHecart weeks ago, this thing is currently broken (the modification is not transmitted to the `PublisherState` stored in session, only the creation destination matter).
There was maybe another reason, but I forgot. @OlivierHecart do you remember?